packages/twenty-front/src/modules/views/states/currentViewIdComponentState.ts

```diff
@@ -1,8 +1,10 @@
-import { createComponentState } from '@/ui/utilities/state/component-state/utils/createComponentState';
+import { createComponentStateV2 } from '@/ui/utilities/state/component-state/utils/createComponentStateV2';
+import { ViewScopeInternalContext } from '@/views/scopes/scope-internal-context/ViewScopeInternalContext';
 
-export const currentViewIdComponentState = createComponentState<
+export const currentViewIdComponentState = createComponentStateV2<
   string | undefined
 >({
   key: 'currentViewIdComponentState',
   defaultValue: undefined,
+  componentContext: ViewScopeInternalContext,
```

Comment:
I would remove the naming "scope" to only keep "component"
ViewComponentInternalContext

We are coupling this componentState api to components but I think it's fine and simplifies it.
I would only use Component / ComponentId naming everywhere

Reply:
if we want a more complete naming it would be ComponentScopeId but I feel ComponentId is good enough